File: tmdb-core/src/main/java/dev/reuss/tmdb/common/company/ProductionCompany.java
package dev.reuss.tmdb.common.company;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Production company for a TMDB resource.
 *
 * @param id            TMDB company id
 * @param logoPath      company logo path
 * @param name          company name
 * @param originCountry origin country
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record ProductionCompany(
        int id,

        @JsonProperty("logo_path")
        String logoPath,

        String name,

        @JsonProperty("origin_country")
        String originCountry
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/company/ProductionCompany.kt
package dev.reuss.tmdb.common.company

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Production company for a TMDB resource.
 *
 * @property id            TMDB company id
 * @property logoPath      company logo path
 * @property name          company name
 * @property originCountry origin country
 */
@JvmRecord
data class ProductionCompany(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("logo_path")
    val logoPath: String?,
    @all:JsonProperty("name")
    val name: String?,
    @all:JsonProperty("origin_country")
    val originCountry: String?,
) : TmdbModel
